integration/internal/integration_tests/integration_test.go

```diff
@@ -122,3 +121,27 @@ func isSomethingListeningOnPort(hostAndPort string) portChecker {
 		return true
 	}
 }
+
+func CheckAPIServerIsReady(kubeCtl *integration.KubeCtl) {
```

Comment:
Why is this check required?  It seems more like validation of the underlying kubeapi implementation than of the fixture that is managing a kube-apiserver binary.

Reply:
We are porting tests in `k/k` and had an issue there. We took the code from [this test](https://github.com/hoegaarden/kubernetes/blob/287fdef1bd98646bc521f4433c1009936d5cf7a2/hack/make-rules/test-cmd-util.sh#L1524-L1535) and used it in here. This should guard against the same issue re-appearing, even if we change the method by which we check that the APIServer is ready to go.

Reply:
Consider documenting your rationale in a comment so that this knowledge is available to future maintainers.